classlib/src/main/java/org/teavm/classlib/java/net/TURLEncoder.java

```diff
@@ -51,6 +51,15 @@ public static String encode(String s) {
         }
         return buf.toString();
     }
+    
+    public static String encode(String s, Charset enc) {
```

Comment:
This approach to implement this method is wrong. The primary logic should be in this method, while the former one (that which takes charset name) should delegate to this method.

Reply:
I see your point, done. Note that I could not add a new test in `URLEncoderTest.java`, as this would make the test itself incompatible with Jaba 8 (which is currently the baseline for this project). However, the changed code is still tested via the existing test cases for `encode(String, String)`.